### ISSUE TITLE
Turning off Teacher Forcing in decoders of Seq2Seq models 

### ISSUE THREAD

**Issue by malsulaimi** [tensorflow/tensorflow#21454]
Have I written custom code:N/A
OS Platform and Distribution :macOS Sierra 10.12.6
TensorFlow installed from : tensforflow.org
TensorFlow version: TensorFlow Version: 1.3.0
Bazel version: NA
CUDA/cuDNN version:NA
GPU model and memory:NA
Exact command to reproduce:NA
Mobile device:NA

Hello , 

I'm posting this request here however I'm not sure if this is a new feature request or it is something already doable in tensorflow but I can not find find an example or any documentation about it  . 

Geting to the Point . I'm experimenting with  seq2seq models . I have followed all the examples available and all is good. Now my model uses Teacher forcing ( passing the true output to the decoder network during training ) and nI  would like to turn it off to see how the model performs without it . unfortunately I do not see any possible way to do it without making my own Helper . I tried to use the inferenceHelper during the Training instead of the TraningHelper. Since the inferenceHelper does not require the true output but the model gives a run time error after some epochs where inferenceHelper is returning predictions with shape other than the expected. my guess is that I inferenceHelper is not meant to be used during training ( as the name suggests) . 

is there a way to turn off the Teacher Forcing , if not are there any plans to incorporate this in Tensorflow ? 

my original decoder network code : 

```
`def decoding_layer_train(encoder_state, dec_cell, dec_embed_input, 
                     target_sequence_length, max_summary_length, 
                     output_layer, keep_prob):
"""
Create a decoding layer for training
:param encoder_state: Encoder State
:param dec_cell: Decoder RNN Cell
:param dec_embed_input: Decoder embedded input
:param target_sequence_length: The lengths of each sequence in the target batch
:param max_summary_length: The length of the longest sequence in the batch
:param output_layer: Function to apply the output layer
:param keep_prob: Dropout keep probability
:return: BasicDecoderOutput containing training logits and sample_id
"""

training_helper = tf.contrib.seq2seq.TrainingHelper(inputs=dec_embed_input,
                                                    sequence_length=target_sequence_length,
                                                    time_major=False)

training_decoder = tf.contrib.seq2seq.BasicDecoder(dec_cell, training_helper, encoder_state, output_layer)

training_decoder_output = tf.contrib.seq2seq.dynamic_decode(training_decoder,
                                                            impute_finished=True,
                                                            maximum_iterations=max_summary_length)[0]
return training_decoder_output
````

thank you

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by malsulaimi** [comment]
Have I written custom code:N/A
OS Platform and Distribution :macOS Sierra 10.12.6
TensorFlow installed from : tensforflow.org
TensorFlow version: TensorFlow Version: 1.3.0
Bazel version: NA
CUDA/cuDNN version:NA
GPU model and memory:NA
Exact command to reproduce:NA
Mobile device:NA

**Comment by malsulaimi** [comment]
is there anyone looking at this ?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 32 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 47 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 62 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 92 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 107 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by superfan123** [comment]
'turning off' will means to change the 'graph'. So you may need to manually get the weight and calculate by yourself. I can't figure this out either. I want to calculate the validation error during training

**Comment by sachinprasadhs** [comment]
Since tf.contrib has been depreciated, seq2seg has been moved under Tensorflow-Addons, you need to install tensorflow addons using 
`pip install tensorflow-addons`
and import the addons like below.
```
import tensorflow as tf
import tensorflow_addons as tfa
```
Refer to the `seq2seq` documentation [here](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq) for the classes which would suite your requirement.
If you still face the issue, open new issue under https://github.com/tensorflow/addons.
